akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -0,0 +1,386 @@
+/**
+ *  Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.dispatch.affinity
+
+import java.lang.invoke.{ MethodHandle, MethodHandles }
+import java.lang.invoke.MethodType.methodType
+import java.util
+import java.util.Collections
+import java.util.concurrent.TimeUnit.MICROSECONDS
+import java.util.concurrent._
+import java.util.concurrent.atomic.AtomicInteger
+import java.util.concurrent.locks.{ Lock, LockSupport, ReentrantLock }
+
+import akka.dispatch._
+import akka.dispatch.affinity.AffinityPool._
+import akka.util.Helpers.Requiring
+import com.typesafe.config.Config
+
+import scala.annotation.tailrec
+import scala.collection.JavaConversions._
```

Comment:
use `JavaConverters` instead of `JavaConversions` to avoid invisible implicit magic, in performance sensitive code we shouldn't use any conversations at all. Here this is used when calling the ConcurrentHashMap and that generates extra allocations (sorry for nitpicking on the allocations, but they matter for overall application performance in code like this, might not show in benchmarks)